src/layer.js

```diff
@@ -148,6 +155,27 @@ export class MapLayer extends HTMLElement {
           this.opacity = newValue;
         }
       break;
+      case 'src':
+        if (!oldValue && this._layer) {
+          // attach a shadowroot if there was no src attribute before
+          if (!this.shadowRoot) {
+            this.attachShadow({mode: 'open'});
+          }
+        } else if (oldValue !== newValue && this._layer) {
+          // re-load the layer element when the src attribute is changed
+          let oldOpacity = this.opacity;
+          // go through the same sequence as if the layer had been removed from
+          // the DOM and re-attached with a new URL source.
+          this.disconnectedCallback();
+          var base = this.baseURI ? this.baseURI : document.baseURI;
+          this._layer = M.mapMLLayer(newValue ? (new URL(newValue, base)).href: null, this);
+          this._layer.on('extentload', this._onLayerExtentLoad, this);
+          this._setUpEvents();
+          if (this.parentNode) {
```

Comment:
perhaps this.isConnected ?  I don't know if there might be a this.parentNode and the whole tree is not connected.